src/internal_modules/roc_fec/block_writer.cpp

```diff
@@ -121,7 +121,7 @@ status::StatusCode BlockWriter::write(const packet::PacketPtr& pp) {
 
     if (!alive_) {
         // TODO(gh-183): return StatusDead
-        return status::StatusOK;
+        return status::StatusNoMem;
```

Comment:
Please change it to StatusAbort, similar to BlockReader. Setting alive_ to false means that we need to abort the session. StatusDead is the outdated name for StatusAbort, so please also remove TODO comment.

We'll also need to modify tests. E.g. currently you have this:

```cpp
LONGS_EQUAL(status::StatusNoMem, writer.write(generate_packet(sn++)));
```

and we would need something like this:

```
LONGS_EQUAL(i == 0 ? status::StatusNoMem : status::StatusAbort,
                    writer.write(generate_packet(sn++)));
```

i.e. we check that the first call returned error, and all subsequent calls return StatusAbort (because alive became false).